sol/private/common.bzl

```diff
@@ -0,0 +1,28 @@
+"""Shared functionality used by one or more rules."""
+
+load("//sol:providers.bzl", "SolRemappingsInfo")
+
+def transitive_remappings(ctx, extra_remappings = {}):
```

Comment:
opt: everywhere this is used, it's wrapped in a SolRemappingsInfo. Maybe it wants to be a factory function that creates SolRemappingsInfo, taking deps into account, similar to the factory function for js_info: https://github.com/aspect-build/rules_js/blob/main/js/private/js_info.bzl#L18
Then at the use sites, you'd construct a SolRemappingsInfo, and you can still iterate over the dict to format the entries as command-line args or a remappings.txt file

Reply:
Good catch, thank you. Done.